internal/wrapper/githubsearchapi.go

```diff
@@ -0,0 +1,81 @@
+package wrapper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+var (
+	SearchApiBaseUrl = "https://api.github.com/search"
+)
+
+type GithubSearchCodeResult struct {
+	FileURL string
+	Repo    string
+}
+
+type GithubSearchApiClient struct {
+	Token   string
+	Client  *http.Client
+	BaseURL string
+}
+
+func NewGithubSearchApiClient(token string) *GithubSearchApiClient {
+	return &GithubSearchApiClient{
+		Token:   token,
+		Client:  &http.Client{},
+		BaseURL: SearchApiBaseUrl,
+	}
+}
+
+func (g *GithubSearchApiClient) SearchCode(ctx context.Context, term, user string) ([]GithubSearchCodeResult, error) {
+	query := term
+	if user != "" {
```

Comment:
what if term or user contains illegal characters for github search ? might your request will get rejected